### PR TITLE
Add ext_address and ext_balance calls

### LINKED ISSUE #58: Implement `ext_address`
Add the ability to call `ext_address` through the pDSL.

### PR DESCRIPTION
**Pull request by holygits**
Closes #58 and #59

### CODE REVIEW

**Comment by holygits** [comment]
Also a sneaky spell fix and cargo-fmt to match the travis CI output

File: core/src/env/srml/srml_only/impls.rs [hunk 1]
```diff
@@ -91,14 +92,9 @@ where
...
-where
-    T: EnvTypes,
-    <T as EnvTypes>::AccountId: for<'a> TryFrom<&'a [u8]>,
-    <T as EnvTypes>::Hash: for<'a> TryFrom<&'a [u8]>,
-{
-    fn caller() -> <Self as EnvTypes>::AccountId {
-        unsafe { sys::ext_caller() };
+impl<T: EnvTypes> SrmlEnv<T> {
+    /// Load the contents of the scratch buffer
+    fn read_scratch_buffer() -> Vec<u8> {
```
Comment (Robbepop) [inline core/src/env/srml/srml_only/impls.rs]: Thanks for the clean up!

File: core/src/env/test_env.rs [hunk 2]
```diff
@@ -205,6 +221,16 @@ impl TestEnvData {
         self.expected_return = expected_bytes.to_vec();
     }
 
+    /// Sets the contract address for the next contract invocation.
+    pub fn set_address(&mut self, new_address: srml::AccountId) {
```
Comment (Robbepop) [inline core/src/env/test_env.rs]: I think it is missing a redirection from `env/test`.

File: core/src/env/test_env.rs [hunk 3]
```diff
@@ -205,6 +221,16 @@ impl TestEnvData {
...
+        self.address = new_address;
+    }
+
+    /// Sets the contract balance for the next contract invocation.
+    pub fn set_balance(&mut self, new_balance: srml::Balance) {
```
Comment (Robbepop) [inline core/src/env/test_env.rs]: same

**Comment by Robbepop** [comment]
Thank you for the PR!

**Pull request merged**